### PR TITLE
Fix Simulator Workflow bugs

### PR DESCRIPTION
**Pull request by Champers5000**
# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [x] Software feature has test associated with it. 
- [ ] Software feature has documentation for it updated.
- [x] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [x] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [x] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

**Review by MichaelMohn — APPROVED** [review]
Looks good

**Review by ray4477 — DISMISSED** [review]
:100:

**Review by connorl309 — DISMISSED** [review]

**Comment by manthanand** [comment]
How did you verify that simulator fails are reached on github?

**Review by MichaelMohn — DISMISSED** [review]

File: Tests/Test_BatteryBalancingSim.c [hunk 1]
```diff
@@ -29,7 +29,7 @@ int main() {
         minions[i].config.tx_data[4] &= 0x00;
     } 
     
-    minions[3].config.tx_data[4] |= 0x01;
+    minions[2].config.tx_data[4] |= 0x01;
```
Comment (manthanand) [inline Tests/Test_BatteryBalancingSim.c:32]: This should just be NUM_MINIONS - 1

File: Validation/Simulator_Tests.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+ulimit -r unlimited
+
+#set -e # to exit bash script upon any commmand failing
+
+yell() { echo "$0: $*" >&2; }
+die() { yell "$*"; exit 111; }
+try() { "$@" || die "$* has failed"; }
```
Comment (manthanand) [inline Validation/Simulator_Tests.sh:7]: write comments explaining what this means.

File: Validation/verify_test.py [hunk 3]
```diff
@@ -94,7 +94,10 @@ def main():
     
     #run the simulator 
     print("Running test " + test_name)
-    os.system("./bps-simulator.out "+ test_name)
+    status = os.system("./bps-simulator.out "+ test_name)
+    if(os.WEXITSTATUS(status) != 0):
```
Comment (manthanand) [inline Validation/verify_test.py:98]: get rid of status variable

File: BSP/Simulator/Src/Simulator.c [hunk 4]
```diff
@@ -265,7 +267,7 @@ void Simulator_Init(char *jsonPath) {
     mkdir(outputdir, S_IRWXU);
 
     // create the log file
-    simulatorLog = open(filename, O_CREAT | O_WRONLY, 0664);
+    simulatorLog = creat(filename, 0664);
```
Comment (manthanand) [inline BSP/Simulator/Src/Simulator.c:270]: write a comment explaining what 0664 is. we don't like magic numbers.

**Review by manthanand — APPROVED** [review]

**Pull request merged**